splinepy/utils/data.py

```diff
@@ -489,7 +490,21 @@ def as_vertex_data(self, resolutions=None, on=None):
         # if resolutions is specified, this is not a location query
         if resolutions is not None:
             if self.has_function:
-                return self.function(self.data, resolutions=resolutions)
+                # enable to have a tuple of splines in the data field
+                # to enable comparisons, enforce para_dim match
+                if isinstance(self.data, tuple):
+                    para_dim = self.data[0].para_dim
+                    if any(d.para_dim != para_dim for d in self.data):
+                        raise ValueError(
+                            "All splines in the data field need to be "
+                            "of the same parametric dimension."
+                        )
+                else:
+                    para_dim = self.data.para_dim
+                return self.function(
+                    self.data,
+                    resolutions=_enforce_len(resolutions, para_dim),
+                )
```

Comment:
https://github.com/tataratat/splinepy/issues/291 would be required here

Reply:
Can we maybe just link this part of the code in the issue, so it will get replaced, once this issue is resolved? This PR is without progress for a long time...